### ISSUE TITLE
Option to disable rich editor button

### ISSUE THREAD

**Issue by PowerUser64** [nextcloud/notes-android#1819]
I think the option to use the rich editor is neat, but I don't see myself needing to use it. Could a setting be added to disable the floating button in the basic editor? Another solution would be to have a setting that moves the button to the ellipsis menu in the upper right. This button bothers me because I often push it by accident and switching back to the basic editor often takes a fair amount of time.

**Comment by cheywood** [comment]
Agreed. I also wouldn't have thought it was something people would want to toggle so frequently that it warrants an action button? As in, wouldn't those wanting a rich text editor just leave it enabled all the time?

**Comment by PowerUser64** [comment]
It would be far more helpful if this button could instead be used to toggle between edit and view mode or something.

**Comment by TheDaAndy** [comment]
I also don't use it, as I often have Internet disabled/loose connection. But without, the rich text editor fails to load. I also often fall into (server/client version conflicts) but I am the only user on the cloud and the changes happened whil using the rich text editor and switching back when it fails. It makes offline usage uncomfortable.
so I would appreciate to remove the action button as well, as it overlaps often my content.

**Comment by cheywood** [comment]
> It would be far more helpful if this button could instead be used to toggle between edit and view mode or something.

True, although I find the button getting in the way of the text itself (when one is writing at the end of the buffer just above the keyboard) kind of makes its existance hard to justify

**Comment by timunrau** [comment]
Just chiming in - I too have the issues of frequently hitting the rich text editor button by accident and it hiding my content. Both are very disturbing. Maybe it can be in the toolbar or as others have mentioned in the 3 dots menu. Personally I don't use the rich text editor as it messes with my markdown formatting.

**Comment by AceJase** [comment]
Same here - the floating button is an annoyance, I'd imagine even to those who do use the rich editor. I'd very much like to be able to remove that as my notes needs are very basic, which was always the draw to this app. It's a nice feature for some I'm sure, but just having the option to disable rich editing or at least just move that floating button to an option in the menu would be enough. Thank you.

**Comment by PowerUser64** [comment]
@stefan-niedermann and/or @AndyScherzinger - what's the current status of this? I see that there are thumbs up's on the original comment from many of the project's devs, and there's an unmerged pull request that fixes this. I don't mean to sound pushy, but I (and others) would really like to see this change implemented, either through merging the pr #1904 or another way. This is an everyday application for most of us (it is for me at least), and this button gets in the way often. I realize that this is an open-source project, you likely have other obligations, and this project is not your first priority. Thank you all for putting this project first where you can, as it has helped many people stop using big tech and enjoy using open-source software!

**Comment by PowerUser64** [comment]
Thank you so much!

**Comment by timunrau** [comment]
I just excitedly upgraded to 4.2.1 in anticipation of this feature being implemented. But I don't see the option to hide the rich text FAB anywhere. Am I missing something?

**Comment by AndyScherzinger** [comment]
Yes, I had to revert it for 4.2.1 since it doesn't fully work unfortunately. There are situation not yet clear to me/us where there is a button and from there onwards you can hit the FAB but nothing happens, so the original PR I tested and merged seems to need some more love to actually be working.

**Comment by timunrau** [comment]
> Yes, I had to revert it for 4.2.1 since it doesn't fully work unfortunately. There are situation not yet clear to me/us where there is a button and from there onwards you can hit the FAB but nothing happens, so the original PR I tested and merged seems to need some more love to actually be working.

Thanks for the update. Hoping for a solution soon!

**Comment by c-o-m-m-a-n-d-e-r-d-a-t-a** [comment]
Plus one

It gave me a grim smile when I looked up this project that its description is 'distraction free'. Sometimes you simply cannot place the caret because the button is in the way. Um...

Otherwise what a fantastically excellent app which I almost live in for my daily reminders and lists.

Keep up the good work!

**Comment by PowerUser64** [comment]
Wouldn't it make more sense to just move the rich editor button to the ellipses menu or settings? I don't think people will need to switch between the two editors so frequently that there needs to be a button in such a prominent place.

**Comment by c-o-m-m-a-n-d-e-r-d-a-t-a** [comment]
For myself at least, I have absolutely no reason to switch to rich edit on my phone. The phone screen's too small for rich editing to make sense anyway. Anything that gets it off the screen, right where you're trying to type, would be a fix.

**Comment by juliushaertl** [comment]
Reopening this one instead of https://github.com/nextcloud/notes/issues/1293 as the PR was reverted

**Comment by PowerUser64** [comment]
Is there any chance this could be added to the 4.4.0 milestone?

**Comment by PowerUser64** [comment]
3 out of 4 of [the top-voted issues](https://github.com/nextcloud/notes-android/issues?q=+is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) in this repository are related to this issue (#1754, #2272, and this one). It would be nice to hear some kind of activity around this, since it's clearly something a lot of people care about.


![Image](https://github.com/user-attachments/assets/c6ff08d4-7acf-4491-b6f0-dff9fd75c1b2)

 Are there any updates around this that I'm not aware of?

**Comment by cheywood** [comment]
> Are there any updates around this that I'm not aware of?

The solution for me was to disable "Direct Edit" in the preferences. This seems to hide the button.

**Comment by PowerUser64** [comment]
You're right! That does seem to hide the button. Thanks for pointing that out!

I think this issue can be closed now, but I'm going to leave that up to the devs in case they want to do something else with it.

Thanks @cheywood!

**Comment by joshtrichards** [comment]
Added in #1904
